src/data/polynomial/eval.lean

```diff
@@ -399,6 +399,8 @@ by rwa [is_root, eval, eval₂_eq_zero_of_dvd_of_eval₂_eq_zero _ _ hpq]
 
 lemma not_is_root_C (r a : R) (hr : r ≠ 0) : ¬ is_root (C r) a := by simpa using hr
 
+lemma eval_surjective (x : R) : function.surjective $ eval x := λ y, ⟨C y, eval_C⟩
```

Comment:
Possibly a nicer proof:
```suggestion
lemma eval_surjective (x : R) : function.surjective $ eval x :=
function.right_inverse.surjective $ λ y, eval_C
```

Reply:
This just makes the lemma two lines, and the original proof is easy enough to understand.

Reply:
Arguably `eval_C` should take its argument explicitly, which would make my suggestion a single function application; but since it doesn't, what you have is ok...